backend/sirius-web-compatibility/src/main/java/org/eclipse/sirius/web/compat/diagrams/EdgeMappingConverter.java

```diff
@@ -130,38 +123,28 @@ public EdgeDescription convert(EdgeMapping edgeMapping) {
         // @formatter:on
     }
 
-    private Function<VariableManager, Optional<Label>> createLabelProvider(LabelStyleDescriptionConverter labelStyleDescriptionConverter, BasicLabelStyleDescription siriusBasicLabelStyleDescription,
-            String idSuffix) {
-        return variableManager -> {
-            return Optional.ofNullable(siriusBasicLabelStyleDescription).map(siriusLabelStyleDescription -> {
-                UUID ownerId = variableManager.get(LabelDescription.OWNER_ID, UUID.class).orElse(null);
-
-                String labelId = ownerId.toString() + idSuffix;
-                String labelExpression = Optional.ofNullable(siriusLabelStyleDescription).map(BasicLabelStyleDescription::getLabelExpression).orElse(""); //$NON-NLS-1$
-                LabelStyleDescription labelStyleDescription = labelStyleDescriptionConverter.convert(siriusLabelStyleDescription);
-
-                // @formatter:off
-                LabelStyle style = LabelStyle.newLabelStyle()
-                        .color(labelStyleDescription.getColorProvider().apply(variableManager))
-                        .fontSize(labelStyleDescription.getFontSizeProvider().apply(variableManager))
-                        .bold(labelStyleDescription.getBoldProvider().apply(variableManager))
-                        .italic(labelStyleDescription.getItalicProvider().apply(variableManager))
-                        .strikeThrough(labelStyleDescription.getStrikeThroughProvider().apply(variableManager))
-                        .underline(labelStyleDescription.getUnderlineProvider().apply(variableManager))
-                        .iconURL(labelStyleDescription.getIconURLProvider().apply(variableManager))
-                        .build();
-
-                return Label.newLabel(labelId)
-                        .type("label:inside-center") //$NON-NLS-1$
-                        .text(new StringValueProvider(this.interpreter, labelExpression).apply(variableManager))
-                        .position(Position.UNDEFINED)
-                        .size(Size.UNDEFINED)
-                        .alignment(Position.UNDEFINED)
-                        .style(style)
-                        .build();
-                // @formatter:on
-            });
-        };
+    private Optional<LabelDescription> createLabelDescription(LabelStyleDescriptionConverter labelStyleDescriptionConverter, BasicLabelStyleDescription siriusBasicLabelStyleDescription,
+            String idSuffix, EdgeMapping edgeMapping) {
+        return Optional.ofNullable(siriusBasicLabelStyleDescription).map(siriusLabelStyleDescription -> {
+            String labelExpression = Optional.ofNullable(siriusLabelStyleDescription).map(BasicLabelStyleDescription::getLabelExpression).orElse(""); //$NON-NLS-1$
```

Comment:
Could you please only call this method if you have what you need from the Sirius description and not propagate all those optionals in the method.